test/hotspot/jtreg/serviceability/tmtools/jstack/DaemonThreadTest.java

```diff
@@ -66,7 +66,15 @@ public static void main(String[] args) throws Exception {
     }
 
     private static void testNoDaemon() throws Exception {
-        testThread(new NormalThread(), "");
+        Thread t = new NormalThread();
+        if (Thread.currentThread().isVirtual()) {
+            // If the NormalThread was created by a current virtual thread
+            // then the NormalThread too will be marked daemon (since virtual threads
+            // are always daemon). In that case, we explicitly reset daemon to false
+            // for the newly created thread.
+            t.setDaemon(false);
+        }
+        testThread(t, "");
```

Comment:
I think the following is a bit clearer:

```
// Virtual threads are always daemon threads. Therefore if the current thread is a
// virtual thread, then NormalThread will be a daemon thread and we need to explicitly
// make it not a daemon thread.
```

You don't actually need the isVirtual() check, especially with the comment explaining things. However, you could also forgo all this logic and the comment by just sticking setDaemon(false) in the NormalThread constructor.

Reply:
Done - I've updated the PR to just call `setDaemon(false)`  from the constructor of `NormalThread` instead of this conditional logic.

When I initially added this logic, I was trying to avoid doing any changes that could impact the testing when running under platform threads. But thinking about this test's context, it merely wants to verify that the jstack output correctly presents daemon/non-daemon status, so explicitly setting a "normal" thread to non-daemon shouldn't impact what this test was originally verifying.

Reply:
Yes, I also briefly wondered if somehow this would be subverting the testing of whether the thread was not daemon by default, but that's not what is being tested here.